File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2024-05-01/examples-js/ApiManagementCreateTagApiLink.js
const { ApiManagementClient } = require("@azure/arm-apimanagement");
const { DefaultAzureCredential } = require("@azure/identity");
require("dotenv/config");

/**
 * This sample demonstrates how to Adds an API to the specified tag via link.
 *
 * @summary Adds an API to the specified tag via link.
 * x-ms-original-file: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2024-05-01/examples/ApiManagementCreateTagApiLink.json
 */
async function apiManagementCreateTagApiLink() {
  const subscriptionId =
    process.env["APIMANAGEMENT_SUBSCRIPTION_ID"] || "00000000-0000-0000-0000-000000000000";
  const resourceGroupName = process.env["APIMANAGEMENT_RESOURCE_GROUP"] || "rg1";
  const serviceName = "apimService1";
  const tagId = "tag1";
  const apiLinkId = "link1";
  const parameters = {
    apiId:
      "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/apis/echo-api",
  };
  const credential = new DefaultAzureCredential();
  const client = new ApiManagementClient(credential, subscriptionId);
  const result = await client.tagApiLink.createOrUpdate(
    resourceGroupName,
    serviceName,
    tagId,
    apiLinkId,
    parameters,
  );
  console.log(result);
}

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2024-05-01/examples-java/ApiManagementCreateTagApiLink.java

/**
 * Samples for TagApiLink CreateOrUpdate.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2024-05-01/examples/
     * ApiManagementCreateTagApiLink.json
     */
    /**
     * Sample code: ApiManagementCreateTagApiLink.
     * 
     * @param manager Entry point to ApiManagementManager.
     */
    public static void
        apiManagementCreateTagApiLink(com.azure.resourcemanager.apimanagement.ApiManagementManager manager) {
        manager.tagApiLinks().define("link1").withExistingTag("rg1", "apimService1", "tag1").withApiId(
            "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/apis/echo-api")
            .create();
    }
}
